OSDOCS-9725 ALBO Install Doc Procedure Updates

jmanthei opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.12+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: https://issues.redhat.com/browse/OSDOCS-9725
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: https://73583--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/aws_load_balancer_operator/install-aws-load-balancer-operator.html
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

jmanthei commented:
4.12, 4.13 and 4.14 have nuanced differences. This is only for 4.14.

jmanthei commented:
@ShudiLi can you QE check the CLI directions please

ShudiLi commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -31,3 +31,62 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
+====
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: aws-load-balancer-operator
+  namespace: aws-load-balancer-operator
+spec:
+  channel: stable-v1.1
+  installPlanApproval: Automatic
+  name: aws-load-balancer-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace                    
+====
+
+[YAML]
+====
+cat <<EOF | oc apply -f
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: aws-load-balancer-operator
+  namespace: aws-load-balancer-operator
+spec:
+  upgradeStrategy: Default
+====
+
+[YAML]
+====
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: aws-load-balancer-operator
+  namespace: aws-load-balancer-operator
+spec:
+  channel: stable-v1.1
+  installPlanApproval: Automatic
+  name: aws-load-balancer-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+EOF
```
looks good to me, but suggest to replace with below:
oc new-project aws-load-balancer-operator

cat <<EOF | oc apply -f -
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  upgradeStrategy: Default
EOF

cat <<EOF | oc apply -f -
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1.1
  installPlanApproval: Automatic
  name: aws-load-balancer-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
EOF

jmanthei replied:
Sure, no problem

jmanthei commented:
/label peer-review-needed

ShudiLi commented:
/lgtm
thanks

JoeAldinger commented:
Direct link for preview: https://73583--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/aws_load_balancer_operator/install-aws-load-balancer-operator

JoeAldinger commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -31,3 +31,46 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
+[source,terminal]
+----
+$ oc -n aws-load-balancer-operator get ip <install_plan_name> --template='{{.status.phase}}{{"\n"}}'
+----
```
These lines for the steps would be:
+ 
The status must be `complete`
+
. Example output
[source,terminal]
----
complete
----

jmanthei replied:
ty

JoeAldinger commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -31,3 +31,46 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
+[source,terminal]
+----
+$ oc -n aws-load-balancer-operator get ip <install_plan_name> --template='{{.status.phase}}{{"\n"}}'
+----
```
```suggestion
+ 
The status must be `complete`
+
. Example output
[source,terminal]
----
complete
----
```

JoeAldinger replied:
I'm not sure why GH is not rendering my comment.

JoeAldinger reviewed (commented):
Hi Jessi, let me know if you have any questions. Big things would be: two modules and working on the procedural steps. Something to think about for the future might be where this should live in the docs. I made a suggestion but maybe asking other networking folks might be a good idea.

michaelryanpeter commented:
@jmanthei If this PR is only for 4.14, please update comment zero in the PR with the accurate version information. Please be aware, PRs merged into `main` must be merged into 4.16. So, if this PR is accurate for 4.14 through 4.16 (4.14+), then you can keep the base branch as `main`.

If this PR should be merged into only OCP 4.14, then you need to change your base branch. The instructions for how to do this is in the OpenShift docs manual: https://docs.google.com/document/u/2/d/e/2PACX-1vRLKWEMHQ3DZroxZKfTu3XcrSdREr6D3oSSayBanEprXhkA2Ciyr2SQuDTYI4aIKUiOPPIQMHgjHeh8/pub#h.5s75lq4bcdl5

jmanthei commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -6,21 +6,21 @@
 . Select the `aws-load-balancer-operator` namespace.
-. Follow the instructions to prepare the Operator for installation.
 . On the *AWS Load Balancer Operator* page, click *Install*.
+. Follow the instructions to prepare the Operator for installation.
```
SME and I did a step by step walkthrough and discovered this was an issue in a followup to a reported doc bug where the instructions were incorrect.

alebedev87 replied:
The instructions are needed only in certain scenarios (STS, UPI):
![image](https://github.com/openshift/openshift-docs/assets/18031474/9756c5cb-c781-4896-881c-60ee776f7221)
In a nominal case - no instructions are needed. However if they are needed they should be done before clicking on `Install` button, the step before runs ahead in this case.

jmanthei commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -31,3 +31,46 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
+
+.Verification
+
+Check the status of an install plan. The status of an install plan must be `Complete`:
```
ty

jmanthei commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -31,3 +31,46 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
+EOF
+====
+
+[YAML]
```
ty

jmanthei commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -31,3 +31,46 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
 
 * Verify that the AWS Load Balancer Operator shows the *Status* as *Succeeded* on the Installed Operators dashboard.
+
+.Installing through the command line
```
check

jldohmann commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,73 @@
+
+.Procedure
+
+. Create aws-load-balancer-operator namespace: oc new-project aws-load-balancer-operator and the created OperatorGroup and Subscription:
```
```suggestion
. Create the `aws-load-balancer-operator` namespace:
[source,terminal]
----
oc new-project aws-load-balancer-operator
----
```

jldohmann reviewed (commented):
hey Jessi, below i've tried to summarize some of the stuff we talked about inline so you can visualize changes. as far as modularization goes, it's looking great! the challenge here is separating out the relevant steps for improved user experience around the creation and application of these configs. hopefully these changes make sense, feel free to ping me if you have questions or want to hop on another call!

jmanthei commented:
@ShudiLi would you please check my edits and confirm the GUI direction change please?

jmanthei commented:
@alebedev87 please confirm my step-by-step breakdown is appropriate and correct

alebedev87 commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -19,8 +17,8 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
 . Navigate to *Operators* → *OperatorHub* in the {product-title} web console.
 . Select the *AWS Load Balancer Operator*. You can use the *Filter by keyword* text box or use the filter list to search for the AWS Load Balancer Operator from the list of Operators.
 . Select the `aws-load-balancer-operator` namespace.
```
This step should be removed. It's mentioned below in the step 6.

alebedev87 replied:
cc @jmanthei

alebedev87 commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,97 @@
+
+.Prerequisites
+
+* You have logged in to the {product-title} web console as a user with `cluster-admin` permissions.
```
The user should be logged to the command line client: `oc`.

alebedev87 commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,97 @@
+oc new-project aws-load-balancer-operator
+----
++
+The following configurations will be automatically created:
```
Do we need to show what will be created before creating? Can't we just show the commands of how to create `OperatorGroup` and `Subscription`? Similar to how we created the namespace above.

jmanthei replied:
I think this question means the breakdown of steps is incorrect at the moment.

alebedev87 commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,97 @@
+[source,terminal]
+Suggested change
+----
+$ oc apply -f 
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+name: aws-load-balancer-operator
+namespace: aws-load-balancer-operator
+spec:
+upgradeStrategy: Default DO WE NEED THE FILENAME HERE
```
This can be replaced by an "in-place" command without any additional file:
```
cat <<EOF | oc apply -f -
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  targetNamespaces: []
EOF
```

jmanthei replied:
Per our standards, we do one step at a time. :)

alebedev87 replied:
Then you can split it in 2 commands:
```bash
cat <<EOF > filename.yaml
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  targetNamespaces: []
EOF
```

```bash
$ oc apply -f filename.yaml
```

ahardin-rh replied:
@alebedev87 Per our style guidelines, we avoid `cat` commands. Is there a way to rework this to only use `oc` commands? Thanks!

alebedev87 replied:
Not really. I think we have to show the yaml and ask to save it into a file:

```yaml
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  targetNamespaces: []
```

The `oc apply -f <filename>` remains the same.

alebedev87 commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,97 @@
++
+[source,terminal]
+----
+$ oc apply -f <name_of_Subscription_config>
```
Similar one line command can be used for the creation of `Subscription`:
```
cat <<EOF | oc apply -f -
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  name: aws-load-balancer-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
EOF
```

jmanthei replied:
We like to break up commands into separate steps to be clear about what each step/line of code does, per our style guide.

alebedev87 replied:
You can do this then:

```bash
cat <<EOF > filename.yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  name: aws-load-balancer-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
EOF
```

```bash
oc apply -f filename.yaml
```

alebedev87 replied:
Here is the the yaml, in case it's the way to go:
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  name: aws-load-balancer-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
```

jmanthei commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -6,21 +6,21 @@
 [id="nw-installing-aws-load-balancer-operator_{context}"]
 = Installing the AWS Load Balancer Operator
```
It really doesn't make sense to put the install modules in the understanding assembly, as the install stuff is procedure modules but the understanding is more informational.

ShudiLi commented:
In  https://73583--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/aws_load_balancer_operator/install-aws-load-balancer-operator, it mentioned "Installation mode as A specific namespace on the cluster", the  Installation mode should be "All namespaces on the cluster (default)", for "A specific namespace on the cluster" mode is not supported by this Operator

ShudiLi commented:
I list the CLI commands and the yaml files to create the ALBO, in the PR, seemed missing CredentialsRequest for creating ALBO
1. oc apply -f namespace.yaml
%cat namespace.yaml 
apiVersion: v1
kind: Namespace
metadata:
  name: aws-load-balancer-operator
2. oc apply -f credentialsrequest.yaml
%cat credentialsrequest.yaml 
apiVersion: cloudcredential.openshift.io/v1
kind: CredentialsRequest
metadata:
  name: aws-load-balancer-operator
  namespace: openshift-cloud-credential-operator
spec:
  providerSpec:
    apiVersion: cloudcredential.openshift.io/v1
    kind: AWSProviderSpec
    statementEntries:
      - action:
          - ec2:DescribeSubnets
        effect: Allow
        resource: "*"
      - action:
          - ec2:CreateTags
          - ec2:DeleteTags
        effect: Allow
        resource: arn:aws:ec2:*:*:subnet/*
      - action:
          - ec2:DescribeVpcs
        effect: Allow
        resource: "*"
  secretRef:
    name: aws-load-balancer-operator
    namespace: aws-load-balancer-operator
  serviceAccountNames:
    - aws-load-balancer-operator-controller-manager
3. oc apply -f operatorgroup.yaml
% cat operatorgroup.yaml 
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: aws-lb-operatorgroup
  namespace: aws-load-balancer-operator
spec:
  upgradeStrategy: Default
4. oc apply -f subscription.yaml
% cat subscription.yaml 
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: aws-load-balancer-operator
  namespace: aws-load-balancer-operator
spec:
  channel: stable-v1
  installPlanApproval: Automatic
  name: aws-load-balancer-operator
  source: qe-app-registry
  sourceNamespace: openshift-marketplace
5. check the installPlan's status.phase
oc -n aws-load-balancer-operator get sub aws-load-balancer-operator --template='{{.status.installplan.name}}{{"\n"}}'
oc -n aws-load-balancer-operator get ip <install_plan_name> --template='{{.status.phase}}{{"\n"}}'

jldohmann commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,134 @@
+oc apply -f namespace.yaml
+cat namespace.yaml
+----
+
```
```suggestion
+
```

jldohmann commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,134 @@
+----
+
+Edit the namespace to include:
+
```
```suggestion
+
```

jldohmann commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,134 @@
+oc apply -f credentialsrequest.yaml
+cat credentialsrequest.yaml
+----
+
```
```suggestion
+
```

jldohmann commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,134 @@
+----
+
+Edit the file to include:
+
```
```suggestion
+
```

jldohmann commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,134 @@
+----
+
+. Create the `subscription` file:
+[source,terminal]
```
```suggestion
+
[source,terminal]
```

jldohmann commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,134 @@
+.Verification
+
+Check the status of the install plan by running the following command:
+
```
```suggestion
+
```

jldohmann commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,134 @@
+oc -n aws-load-balancer-operator get sub aws-load-balancer-operator --template='{{.status.installplan.name}}{{"\n"}}'
+oc -n aws-load-balancer-operator get ip <install_plan_name> --template='{{.status.phase}}{{"\n"}}'
+----
+
```
```suggestion
+
```

jldohmann reviewed (commented):
the `+` delimiter is necessary to make sure the ordered lists in the procedure are handled correctly by asciidoctor

jldohmann reviewed (approved):
nice work! 😄 just a quick check: things look good to me

jldohmann commented:
oh, i just saw one of your earlier comments: if this is for 4.14 only, you will need to change the target branch you're merging into; merging into `main` requires merging into `enterprise-4.16` branch. however, if the changes between them are small, you can open up a second PR to make the tweaks for the respective branches instead. up to you!

bergerhoffer commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -19,15 +17,14 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
 . Select the `aws-load-balancer-operator` namespace.
-. Follow the instructions to prepare the Operator for installation.
-. On the *AWS Load Balancer Operator* page, click *Install*.
+. (Optional) Follow the instructions to prepare the Operator for installation in custom scenarios when you want to use a security token service (STS) or a User Provisioned Infrastructure (UPI).
```
Per IBM Style, use "Optional: " to indicate an optional step (see https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps)

We also don't use the abbreviations UPI and IPI in the docs and do hyphenated lowercase (see https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#user-provisioned-infrastructure)

```suggestion
. Optional: Follow the instructions to prepare the Operator for installation in custom scenarios when you want to use a security token service (STS) or user-provisioned infrastructure.
```

bergerhoffer commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
+:_mod-docs-content-type: PROCEDURE
+[id="nw-installing-aws-load-balancer-operator-cli_{context}"]
+= Installing the AWS Load Balancer Operator with the command line
+
```
Procedure modules should have an intro sentence. See: https://github.com/redhat-documentation/modular-docs/blob/main/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc?plain=1#L25

bergerhoffer commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
+
+* Your cluster is configured with AWS as the platform type and cloud provider.
+
+* You are logged into the command line client `oc`.
```
Should write oc like below. See: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#openshift-cli

```suggestion
* You are logged into the OpenShift CLI (`oc`).
```

jmanthei replied:
ty

bergerhoffer commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
+----
+
+.Verification
++
```
These "+" are rendering in the preview

![image](https://github.com/openshift/openshift-docs/assets/7443887/e30f355a-53d2-4bd9-b1e3-86bce2efc7f2)


Also, there is no step/text before the commands. If it's a single step, it should just be an unnumbered bullet (`*`). But if you make it two steps (because there are two commands), then it would be numbered (`.`).

Then, you would need these "+" in between the step and the code block to keep it together.

bergerhoffer commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
++
+[source,terminal]
+----
+oc -n aws-load-balancer-operator get sub aws-load-balancer-operator --template='{{.status.installplan.name}}{{"\n"}}'
```
There are two commands in this code block. There should only be a single command in each code block. See: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#single-command-per-code-block

There also needs to be a prompt in front of each terminal command, like below, for all of the oc commands.

```suggestion
$ oc -n aws-load-balancer-operator get sub aws-load-balancer-operator --template='{{.status.installplan.name}}{{"\n"}}'
```

bergerhoffer commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
+oc -n aws-load-balancer-operator get ip <install_plan_name> --template='{{.status.phase}}{{"\n"}}'
+----
++
+The output must be `Complete`. 
```
Not critical, but you could consider showing an `.Example output` code block of what the expected output for these commands are. It's not clear, since there are two commands, and I expect one of them is expected to show "Complete". But not sure if both of them are, or just one of them.

bergerhoffer commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
++
+[source, terminal]
+----
+oc apply -f credentialsrequest.yaml
```
Same for all of these to only have a single command in each code block, and making sure to have a "$" prompt.

bergerhoffer commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
+
+. Create the `credentialsrequest` file:
++
+[source, terminal]
```
No space preferred

```suggestion
[source,terminal]
```

bergerhoffer commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
++
+[source,terminal]
+----
+oc apply -f namespace.yaml
```
These commands look to be out of order. This is trying to apply a file to the cluster that hasn't been created yet. Also, the `cat` command doesn't actually do anything other than output the contents of a file that already exists.

We have a lot of examples in the repo with procedures that create a file with certain contents, and then apply the file to the cluster.

There are lots of ways to do it, but the idea should be the same that the file is created first, then applied to the cluster. 

Here are a few examples you could model after:

* https://docs.openshift.com/container-platform/4.15/security/cert_manager_operator/cert-manager-operator-issuer-acme.html#cert-manager-acme-http01_cert-manager-operator-issuer-acme
* https://docs.openshift.com/container-platform/4.15/networking/metallb/about-advertising-ipaddresspool.html
* https://docs.openshift.com/container-platform/4.15/logging/log_storage/installing-log-storage.html

So recommend looking at these and reworking these steps to follow something similar to one of these.

bergerhoffer reviewed (commented):
A few things to fix before this can be merged. I tried to cite as many locations of guidelines as possible, but please let me know if you have any questions on any of it!

bergerhoffer commented:
Same comment also about clarifying which versions this PR is applicable for. It's currently marked for 4.12+. If you don't want it in 4.12 and 4.13 because there are things that need to be changed first, please update it to list only 4.14+, and take care of the other versions in separate PRs.

And also making sure that by your statement of "this is only for 4.14" that you mean 4.14+ (4.14, 4.15, and 4.16). If it really was ONLY for 4.14, then as Jesse mentioned above, this PR would have to be retargeted to only go directly into enterprise-4.14.

alebedev87 commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -19,15 +17,14 @@ You can install the AWS Load Balancer Operator from the OperatorHub by using the
 . Select the `aws-load-balancer-operator` namespace.
-. Follow the instructions to prepare the Operator for installation.
-. On the *AWS Load Balancer Operator* page, click *Install*.
+. (Optional) Follow the instructions to prepare the Operator for installation in custom scenarios when you want to use a security token service (STS) or a User Provisioned Infrastructure (UPI).
```
We have a separate chapter for the STS installation: "Installing the AWS Load Balancer Operator on a cluster using the AWS Security Token Service". Should we drop this step or reference the STS chapter?

jmanthei replied:
Are you talking about https://docs.openshift.com/container-platform/4.15/networking/aws_load_balancer_operator/installing-albo-sts-cluster.html ?

There is no install steps there at the moment, and we probably need to include these install steps at the end of the setup directions.

alebedev87 replied:
> Are you talking about https://docs.openshift.com/container-platform/4.15/networking/aws_load_balancer_operator/installing-albo-sts-cluster.html ?

Yes.

> There is no install steps there at the moment, and we probably need to include these install steps at the end of the setup directions.

I would prefer a link just to avoid redundancy. But I don't know what your conventions about this.

alebedev87 commented on modules/installing-aws-load-balancer-operator-cli.adoc:
```diff
@@ -0,0 +1,133 @@
++
+[source,yaml]
+----
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: aws-load-balancer-operator
+  namespace: openshift-cloud-credential-operator
+spec:
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+      - action:
+          - ec2:DescribeSubnets
+        effect: Allow
+        resource: "*"
+      - action:
+          - ec2:CreateTags
+          - ec2:DeleteTags
+        effect: Allow
+        resource: arn:aws:ec2:*:*:subnet/*
+      - action:
+          - ec2:DescribeVpcs
+        effect: Allow
+        resource: "*"
+  secretRef:
+    name: aws-load-balancer-operator
+    namespace: aws-load-balancer-operator
+  serviceAccountNames:
+    - aws-load-balancer-operator-controller-manager
+----
```
The `CredentialsRequest` resource is only need on the version before 4.14. On OCP 4.14 `CredentialsRequest` is created by the operator.

jmanthei replied:
Cherrypick fix

alebedev87 reviewed (commented):
I had another quick look.

jmanthei commented:
> Edited to correct: previously, versions have had different channels, but recently we updated dev/eng to accomodate for less docs changes. The channels are the same 4.12+ and they were the nuance we were correcting for.
> 

@michaelryanpeter @jldohmann @bergerhoffer I have updated the title and hopefully this comment 1 clears up this issue. No weird cherrypicking for this one. :)

jmanthei commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -4,30 +4,28 @@
 
 * Your cluster is configured with AWS as the platform type and cloud provider.
 
+* If you are using a security token service (STS) or user-provisioned infrastructure, follow the related preparation steps. For example, if you are using AWS Security Token Service, see "Preparing for the AWS Load Balancer Operator on a cluster using the AWS Security Token Service (STS)".
```
@alebedev87 Moved this STS info to the prerequisite section to be more in-line with modularization.
If that's okay,
1. do we have a suggested UPI doc?
2.  should this also be in the CLI procedure's prerequisites then?

jmanthei commented:
/label peer-review-needed

jmanthei commented:
/remove-label peer-review-done

jmanthei commented:
@ShudiLi please review all the changes

mburke5678 commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -4,30 +4,30 @@
 :_mod-docs-content-type: PROCEDURE
 [id="nw-installing-aws-load-balancer-operator_{context}"]
-= Installing the AWS Load Balancer Operator
+= Installing AWS Load Balancer Operator by using the web console
```
I would leave the _the_.

mburke5678 replied:
Especially where you added a _the_ in the [following assembly](https://github.com/openshift/openshift-docs/pull/73583/files#diff-04550ce247b3f955736f564779b31dd454d1295da1e2fb50d07fe37817511ad1R3).

jmanthei replied:
Fixed, thank you

mburke5678 commented:
@jmanthei A few nits. Otherwise LGTM

ShudiLi commented on modules/installing-aws-load-balancer-operator.adoc:
```diff
@@ -4,30 +4,30 @@
-. Follow the instructions to prepare the Operator for installation.
-. On the *AWS Load Balancer Operator* page, click *Install*.
 . On the *Install Operator* page, select the following options:
 .. *Update the channel* as *stable-v1*.
 .. *Installation mode* as *A specific namespace on the cluster*.
 .. *Installed Namespace* as `aws-load-balancer-operator`. If the `aws-load-balancer-operator` namespace does not exist, it gets created during the Operator installation.
```
Installation mode should be "All namespaces on the cluster (default)"

```
Installation mode
All namespaces on the cluster (default)
Operator will be available in all Namespaces.

A specific namespace on the cluster
This mode is not supported by this Operator
```

jmanthei replied:
ty

mburke5678 commented:
/cherrypick enterprise-4.12

mburke5678 commented:
/cherrypick enterprise-4.13

mburke5678 commented:
/cherrypick enterprise-4.14

mburke5678 commented:
/cherrypick enterprise-4.15

mburke5678 commented:
/cherrypick enterprise-4.16
